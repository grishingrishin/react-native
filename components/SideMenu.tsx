import React from 'react';
import { TouchableHighlight, FlatList } from 'react-native';
import styled from 'styled-components/native';

const Wrapper = styled.View`
  position: absolute;
  top: 0;
  left: 0;
  z-index: 999999;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,.3);
`;

const Menu = styled.View`
  width: 80%;
  height: 100%;
  background-color: #fff;
`;

const Header = styled.View`
  padding: 28px 24px;
  border-bottom-width: .5px;
  border-bottom-color: rgba(0,0,0,.3);
`;

const Actions = styled.View`
  margin-bottom: 24px;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

const Closed = styled.Image`
  width: 34px;
  height: 20px;
`;

const Settings = styled.Image`
  width: 24px;
  height: 24px;
`;

const User = styled.View`
  flex-direction: row;
  align-items: center;
`;

const Avatar = styled.Image`
  width: 98px;
  height: 98px;
  border-radius: 120px;
`;

const UserName = styled.Text`
  flex: 1;
  padding: 0 12px;
  font-size: 28px;
  font-family: Roboto-Light;
`;

const List = styled.View`
  padding: 14px 0;
`;

const MenuItem = styled.View`
  padding: 14px 24px;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
`;

const ItemIcon = styled.Image`
  width: 24px;
  height: 24px;
  margin-right: 14px;
`;

const ItemText = styled.Text`
  font-size: 18px;
  font-family: Roboto-Light;
`;

const MENU_ITEMS = [
  {
    id: 'bd7acbea-c1b1-46c2-aed5-3ad53abb28ba',
    icon: '../assets/profile.png',
    title: 'Profile',
  },
  {
    id: '3ac68afc-c605-48d3-a4f8-fbd91aa97f63',
    icon: '../assets/toDoList.png',
    title: 'To Do List',
  },
  {
    id: '58694a0f-3da1-471f-bd96-145571e29d72',
    icon: '../assets/groups.png',
    title: 'Groups',
  },
  {
    id: '7ac68afc-c685-48d3-a4f9-fbd91aa97f65',
    icon: '../assets/news.png',
    title: 'News',
  },
];

const Item = ({ title }: any) => {
  return (
    <TouchableHighlight>
      <MenuItem>
        <ItemIcon
          source={require('../assets/profile.png')}
        />
        <ItemText>{title}</ItemText>
      </MenuItem>
    </TouchableHighlight>
  );
}

const SideMenu = () => {
  const renderItem = ({ item }: any) => (
    <Item
      id={item.id}
      title={item.title}
    />
  );

  return (
    <Wrapper>
      <Menu>
        <Header>
          <Actions>
            <TouchableHighlight>
              <Closed
                source={require('../assets/arrow.png')}
              />
            </TouchableHighlight>
            <TouchableHighlight>
              <Settings
                source={require('../assets/settings.png')}
              />
            </TouchableHighlight>
          </Actions>
          <User>
            <Avatar
              source={require('../assets/sidemenu_avatar.png')}
            />
            <UserName>Sarah Townshend</UserName>
          </User>
        </Header>
        <List>
          <FlatList
            data={MENU_ITEMS}
            renderItem={renderItem}
            keyExtractor={item => item.id}
          />
        </List>
      </Menu>
    </Wrapper>
  )
}

export default SideMenu;