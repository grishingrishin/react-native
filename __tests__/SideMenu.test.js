import React from 'react';
// import {shallow} from 'enzyme';
import {create} from 'react-test-renderer';
import SideMenu from '../components/SideMenu';

describe('<SideMenu />', () => {
  describe('Jest tests', () => {
    it('renders correctly', () => {
      const tree = create(<SideMenu />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});