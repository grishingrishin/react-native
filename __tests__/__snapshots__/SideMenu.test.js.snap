// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideMenu /> Jest tests renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0,0,0,0.3)",
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": 999999,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "height": "100%",
          "width": "80%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingBottom": 28,
            "paddingLeft": 24,
            "paddingRight": 24,
            "paddingTop": 28,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 24,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/arrow.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 20,
                  "width": 34,
                },
              ]
            }
          />
        </View>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/settings.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/sidemenu_avatar.png",
            }
          }
          style={
            Array [
              Object {
                "borderRadius": 120,
                "height": 98,
                "width": 98,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Roboto-Light",
                "fontSize": 28,
                "paddingBottom": 0,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 0,
              },
            ]
          }
        >
          Sarah Townshend
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "borderBottomColor": "rgba(0,0,0,0.3)",
            "borderBottomWidth": 0.5,
            "borderTopColor": "rgba(0,0,0,0.3)",
            "borderTopWidth": 0.5,
            "paddingBottom": 28,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 28,
          },
        ]
      }
    >
      <RCTScrollView
        data={
          Array [
            Object {
              "icon": "../assets/profile.png",
              "id": "bd7acbea-c1b1-46c2-aed5-3ad53abb28ba",
              "title": "Profile",
            },
            Object {
              "icon": "../assets/toDoList.png",
              "id": "3ac68afc-c605-48d3-a4f8-fbd91aa97f63",
              "title": "To Do List",
            },
            Object {
              "icon": "../assets/groups.png",
              "id": "58694a0f-3da1-471f-bd96-145571e29d72",
              "title": "Groups",
            },
            Object {
              "icon": "../assets/news.png",
              "id": "7ac68afc-c685-48d3-a4f9-fbd91aa97f65",
              "title": "News",
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                      "paddingBottom": 18,
                      "paddingLeft": 24,
                      "paddingRight": 24,
                      "paddingTop": 18,
                    },
                  ]
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../assets/profile.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "height": 24,
                        "marginRight": 14,
                        "width": 24,
                      },
                    ]
                  }
                />
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#333333",
                        "fontFamily": "Roboto-Light",
                        "fontSize": 18,
                      },
                    ]
                  }
                >
                  Profile
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                      "paddingBottom": 18,
                      "paddingLeft": 24,
                      "paddingRight": 24,
                      "paddingTop": 18,
                    },
                  ]
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../assets/profile.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "height": 24,
                        "marginRight": 14,
                        "width": 24,
                      },
                    ]
                  }
                />
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#333333",
                        "fontFamily": "Roboto-Light",
                        "fontSize": 18,
                      },
                    ]
                  }
                >
                  To Do List
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                      "paddingBottom": 18,
                      "paddingLeft": 24,
                      "paddingRight": 24,
                      "paddingTop": 18,
                    },
                  ]
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../assets/profile.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "height": 24,
                        "marginRight": 14,
                        "width": 24,
                      },
                    ]
                  }
                />
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#333333",
                        "fontFamily": "Roboto-Light",
                        "fontSize": 18,
                      },
                    ]
                  }
                >
                  Groups
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                      "paddingBottom": 18,
                      "paddingLeft": 24,
                      "paddingRight": 24,
                      "paddingTop": 18,
                    },
                  ]
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../assets/profile.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "height": 24,
                        "marginRight": 14,
                        "width": 24,
                      },
                    ]
                  }
                />
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#333333",
                        "fontFamily": "Roboto-Light",
                        "fontSize": 18,
                      },
                    ]
                  }
                >
                  News
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingBottom": 18,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 18,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingBottom": 18,
                "paddingLeft": 24,
                "paddingRight": 24,
                "paddingTop": 18,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/signout.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "marginRight": 14,
                  "width": 24,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontFamily": "Roboto-Light",
                  "fontSize": 18,
                },
              ]
            }
          >
            Sign out
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
